### ISSUE TITLE
bug(schematcis): Failed ng add because main key doesn't find with @angular-devkit/build-angular:application builder

### ISSUE THREAD

**Issue by yatho** [angular/components#28132]
### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

16

### Description

When I am using ```ng add @angular/material@17```, I have the following error : ```Could not find the project main file inside of the workspace config (src)```.

With Angular 17 when I create a new project with CLI, I use `@angular-devkit/build-angular:application` builder and application builder doesn't have `main` key in `angular.json` but `browser` key instead.

So ng add failed.

### Reproduction

Please do the following command :

- ng new test-app
- ng add @angular/material@17

### Expected Behavior

Ng add works correctly

### Actual Behavior

It is impossible to add material with schematics with application builder.

### Environment

- Angular: 17
- CDK/Material: 17

**Comment by BojanKogoj** [comment]
Is this connected to https://github.com/angular/angular-cli/issues/26284?

**Comment by yatho** [comment]
> Is this connected to https://github.com/angular/angular-cli/issues/26284?

I don't think so. The 26284 issue seems to be a typescript compilation problem. With this one the main.ts is not found with schematics in src folder. I am not sure.
